src/source_sampling.h

```diff
@@ -118,30 +151,82 @@ namespace pyne {
     ///                       number of energy groups as <src_tag_name> or 1.
     ///                       If 1 (i.e. spatial biasing only), all energy groups
     ///                       within a mesh volume element are sampled equally.
-    Sampler(std::string filename, 
-            std::string src_tag_name, 
-            std::vector<double> e_bounds, 
+    Sampler(std::string filename,
```

Comment:
instead of overloading the constuctor, pass the `mode` to the constructor and have it setup correctly.

Reply:
These constructors are not only used by DAG-MCNP, but also by python when performing tests. 
From DAG-MCNP, we can only pass an integer parameter `mode` into the code, all the other file names are hard coded.
But for the PyNE unit tests, we need to change these file names, otherwise we can't make different unit tests to read different test input files. 
I think we should keep these overloading constructors for the purpose of PyNE test usage

Reply:
Meant to send that above comment out yesterday. Sorry for the delay.

Reply:
That's true. We have 6 constructors now, they are in four types of parameter sets. Actually, what we really need to construct is the `filename`, `e_bounds` and made`. `sub_mode` can be derived from `mode`.   
```
Sampler(std::string filename,
              std::vector e_bounds,
              int mode);
```
That means, we need only one constructor. However, this design will absolutely introduce big changes to the `source_sampleing` and replace the original valid constructors for normal r2s. I don't think we should make these changes before we make sure that `sub_voxel` code runs as we expect.